File: 5_LongestPalindromicSubstring/c++/LongestPalindromicSubstring.cpp
class Solution {
public:
    string longestPalindrome(string s) {
        if(s.size()<2){
            return s;
        }
        string res = "";
        string meta = "";
        for(int i = 0; i<s.size(); i++){
            meta= FindLongestPal(s,i,i);
            if(meta.size()>res.size()){
                res = meta;
            }
            meta = FindLongestPal(s,i,i+1);
            if(meta.size()>res.size()){
                res = meta;
            }
        }
        if(res.size() == 0){
            res = s.substr(0,1);
        }
        return res;
        
        
    }
    string FindLongestPal(string s, int s1, int s2){
        string res = "";
        while((s1>=0)&&(s2<s.size())){
            if(s[s1] != s[s2]){
                break;
            }
            else{
                res = s.substr(s1,s2-s1+1);
            }
            s1--;
            s2++;
        }
        return res;
    }
};

File: 5_LongestPalindromicSubstring/c#/LongestPalindromicSubstring.cs
public class Solution {
    public string LongestPalindrome(string s) {
        int maxLength = 0;
        int maxLeft = 0;
        int maxRight = 0;
        for(int meta = 0; meta<s.Length; meta++){
            int left = meta;
            int right = meta;
            while((right<s.Length-1)&&(s[right]==s[right+1])){
                right++;
            }
            while((left>0)&&(right<s.Length-1)&&(s[left-1]==s[right+1])){
                left--;
                right++;
            }
            if(right-left+1>maxLength){
                maxLength = right-left+1;
                maxLeft = left;
                maxRight = right;
            }
        }
        return s.Substring(maxLeft,maxLength);
    }
}